packages/autocomplete-core/src/utils/__tests__/isSamsung.test.ts

```diff
@@ -0,0 +1,317 @@
+import { isSamsung } from '../isSamsung';
+
+describe('isSamsung', () => {
+  describe('Android', () => {
+    describe('Chrome', () => {
+      test('returns true with a Samsung Galaxy S20 (mobile) user agent', () => {
+        expect(
+          isSamsung(
+            'Mozilla/5.0 (Linux; Android 10; SM-G981B) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.162 Mobile Safari/537.36'
+          )
+        ).toEqual(true);
+      });
+
+      test('returns true with a Samsung Galaxy S4 (mobile) user agent', () => {
+        expect(
+          isSamsung(
+            'Mozilla/5.0 (Linux; Android 5.0.1; SAMSUNG-SGH-I337 Build/LRX22C) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.93 Mobile Safari/537.36'
+          )
+        ).toEqual(true);
+      });
+
+      test('returns true with a Samsung Galaxy Tab S3 (tablet) user agent', () => {
+        expect(
+          isSamsung(
+            'Mozilla/5.0 (Linux; Android 7.0; SM-T827R4 Build/NRD90M) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.116 Safari/537.36'
+          )
+        ).toEqual(true);
+      });
+    });
+
+    describe('Firefox', () => {
+      test('returns false with an Android 8 (mobile) user agent', () => {
+        expect(
+          isSamsung(
+            'Mozilla/5.0 (Android 8.0.0; Mobile; rv:98.0) Gecko/98.0 Firefox/98.0'
+          )
+        ).toEqual(false);
+      });
+
+      test('returns false with a Android 9 (tablet) user agent', () => {
+        expect(
+          isSamsung(
+            'Mozilla/5.0 (Android 9; Tablet; rv:97.0) Gecko/97.0 Firefox/97.0'
+          )
+        ).toEqual(false);
+      });
```

Comment:
This tests operating systems, not devices.

Reply:
I don't have much more infos about this device so I indeed used the OS, should I write `with an Android device (mobile) user agent` for instance?

Reply:
Do you know where this user agent comes from? If we don't have a specific device I'm not sure we should test this.

Reply:
I found it on this website: https://useragents.io/explore/devices/types/mobile-phone/maker/mozilla-foundation-757/uas/mozilla-5-0-android-8-0-0-mobile-rv98-0-gecko-98-0-firefox-98-0_5eb8d79c39e29778f32c54253a98a9ea
I tried to find proper user agents from devices on Android Firefox, but it seems hard.

Reply:
Let's do that for both test suites, thanks!